### ISSUE TITLE
DFU upload problem on win32 - DFU ABORT bug

### ISSUE THREAD

**Issue by bnewbold** [leaflabs/maple-ide#10]
_Original author: mbolivar...@gmail.com (November 16, 2010 08:54:14)_

See http://forums.leaflabs.com/topic.php?id=128

_Original issue: http://code.google.com/p/leaflabs/issues/detail?id=8_

**Comment by bnewbold** [comment]
_From ajme...@mit.edu on December 09, 2010 18:50:07_
tldr: DFU is not ABORTING upload transactions after a failed packet, it should though. Why packets are failing is a separate issue, this can be fixed by better DFU error handling. 

It is now evidently clear and reproducible that on some machines, there is a high DFU failure rate, from which Maple does not gracefully recover. During these botched uploads, Maple gets stuck in the dfu statemachine somewhere and thus does not accept attempts to retry DFU. 

The root cause of this is very likely due to some problem with how USB is buffering data, perhaps some data is being lost and breaking the state machine. Debugging this is hard, because the addition of JTAG and the debugger usually nudges the timing in such a way that these errors are hard to reproduce on the testbench. 

Besides fixing the root cause (requiring a large effort into soon to deprecated code), the _right_ solution is to mod the bootloader to gracefully abort failed DFU transactions (which it should do. That it doesnt is a bug). 

This problem may be alleviated by: reducing the transfer speed, reducing the DFU packet size.
